conn.go

```diff
@@ -126,6 +134,37 @@ func DialTLS(network, addr string, config *tls.Config) (*Conn, error) {
 	return conn, nil
 }
 
+// DialURL connects to the given ldap URL vie TCP using tls.Dial or net.Dial if ldaps://
+// or ldap:// specified as protocol. On success a new Conn for the connection
+// is returned.
+func DialURL(addr string) (*Conn, error) {
+
+	lurl, err := url.Parse(addr)
+	if err != nil {
+		return nil, NewError(ErrorNetwork, err)
+	}
+
```

Comment:
Ok. makes sense. However, I cant use net.SplitHostPort directly as it returns an error it port is not provided.

Reply:
taking the error to mean the port wasn't provided (or wasn't valid) is typical

also, use JoinHostPort rather than concatenating with ":" (for ipv6 safety)

Reply:
@liggitt it's already done with latest update